src/password.cpp

```diff
@@ -0,0 +1,45 @@
+#include <fstream>
+#include <iostream>
+#include <stdio.h>
+#include <wordexp.h>
+
+#include "password.h"
+
+const std::string read_password(const char* file) {
+    wordexp_t exp;
+    std::ifstream ifs;
+    std::string pass = "";
+
+    wordexp(file,&exp,0);
+    if (exp.we_wordc > 0) {
+        ifs.open(exp.we_wordv[0]);
+    }
+    wordfree(&exp);
+    if (!ifs.is_open()) {
+    } else {
+        std::getline(ifs, pass);
+    }
+    return pass;
+}
+
+const std::string eval_password(const char* cmd) {
+    char buffer[128];
+    std::string pass = "";
+
+    FILE* pipe = popen(cmd, "r");
+    if (!pipe) {
+    } else {
+        try {
+            while (!feof(pipe)) {
+                if (fgets(buffer, 128, pipe) != NULL) {
+                    buffer[strcspn(buffer, "\r\n")] = 0;
```

Comment:
> It bugs me how hard and error-prone some easy tasks like "read until the first occurrence of (any) newline character" can be in C :(

Write in C++ then :P It'll be just as hard and error-prone as writing in C, but at least you can sometimes distract yourself with shiny features and pretend you're writing in a higher-level language :)

Joking aside, I sometimes entertain myself by musing about the languages I could rewrite Newsbeuter in, because C++ is just too low-level for the things we do here. So far I always ended up deciding it's just too much effort, and the new language isn't as popular as C++ so I'll get less contributions.